forte/data/data_store.py

```diff
@@ -493,6 +495,129 @@ def get_entry(self, tid: int) -> Tuple[List, int, int]:
         # Otherwise, use ``index_id`` to find the index of the entry.
         raise NotImplementedError
 
+    def co_iterator(self, type_names: List[str]) -> Iterator[List]:
+        r"""
+        Given two or more type names, iterate their entry lists from beginning to end together.
```

Comment:
1.  Need to emphasize that `co_iterator` works on annotation-like types. Actually, it is better to reflect that in the function name, maybe rename this to `co_iterator_annotation_like` (In the future we may create iterators for more complex objects like bounding boxes)
2. Back-tip the variable names.

For entries that have the exact same `begin`, `end`, and `type_name`, what's the order? It is fine to say the order is undefined (or arbitrary), but this behavior needs to be mentioned.

Reply:
edited based on the review.